README.md

```diff
@@ -118,7 +118,7 @@ class ExampleTable extends Table
     public function searchManager()
     {
         /** @var \Search\Manager $searchManager */
-        $searchManager = $this->behaviors()->Search->searchManager();
+        $searchManager = $this->behaviors()->get('Search')->searchManager();
```

Comment:
Do we have typehinting or autocomplete on the getter here? Otherwise this is no improvement at all. Magic strings are usually worse then the attributes.
I would be fine with using get(magicstring) + IdeHelper generate of Behaviors::get() etc.

Reply:
Well my motivation to change it was because PHP Storm underlined it as a warning for a magic property.

Reply:
We need to annotate/typehint on the behaviors() collection then, using the meta file etc.
That should fix it for IDEs?

Reply:
Hm, I just checked.. The only way to get autocomplete working for
```php
$searchManager = $this->behaviors()->get('Search')->searchManager();
```

Is to use the following meta map on the ObjectRegistry (unfortunately, the BehaviorRegistry doesnt work):
```php
	override(
		\Cake\Core\ObjectRegistry::get(0),
		map([
			'Search' => \Search\Model\Behavior\SearchBehavior::class,
		])
	);
```

Reply:
```diff
-        $searchManager = $this->behaviors()->Search->searchManager();
+        $searchManager = $this->behaviors()->get('Search')->searchManager();
```

Note that this is probably also slower.

Reply:
Speed is not relevant here imo.
the important thing missing is ide comp.

Reply:
That's the attitude that I am concerned about (not this special case anyway, not a problem).

Reply:
:) Lets talk on slack if you dont understand the difference on this suspect. Please dont shift focus away from what actually matters here.
Your "concerned about" remark was also not neccessary :/

Reply:
@davidyell Back to topic

I got actually the property to work, the magic string getter not..
How about we start using
```php
$searchManager = $this->behaviors()->search()->searchManager();
```
etc?
That would also work to be autocompleted/discovered.